superset-frontend/src/explore/exploreUtils.js

```diff
@@ -248,6 +248,14 @@ export function postForm(url, payload, target = '_blank') {
   document.body.removeChild(hiddenForm);
 }
 
+export function columnCount(columnsLength) {
+  let pageSize;
+  if (columnsLength) {
+    pageSize = Math.ceil(Math.max(5, 10000 / columnsLength));
+  }
+  return pageSize || 50;
+}
```

Comment:
This function could be named more appropriately, e.g. `getDataTablePageSize` or similar.